interview-questions.md

```diff
@@ -8,62 +8,62 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What is the difference between a parameter and an argument?
 
-  Your answer:
+  Your answer: A parameter is the logic used to identify what will be passed into a function or method. The parameter is used to tell the developer what kind of information is expected. The argument is the actual data being used.
 
-  Researched answer:
+  Researched answer: The technical definition is an argument is initialized into the parameter defined. There are input and output parameters. If a parameter that has been initialized was left undefined that is was it will return by default. You can then target the undefined value for a catch-all if needed.
 
 
 
 2. The JavaScript built in method .map() takes predefined parameters. What are they? Which are required and which are optional?
 
-  Your answer:
+  Your answer: The three predefined parameters that the .map() method takes are value, index, and array. Only value is required, and the other two are optional.
 
-  Researched answer:
+  Researched answer: There is a fourth optional parameter for map called thisValue, it is used for a this value inside of the function.
 
 
 
 3. What is the difference between map and filter?
 
-  Your answer:
+  Your answer: The difference between the map and filter built in methods are that map iterates through each element in an array and returns a new array. Filter iterates through each element in an array using a conditional and returns a new array that is shorter.
 
-  Researched answer:
+  Researched answer: Filter returns a subset of the given array, meaning it is not a new array.  Map does something to each element in an array while filter makes a decision based on each element inside of the array.
 
 
 
 4. What is the difference between a function and a method?
 
-  Your answer:
+  Your answer: The difference between function and method are that methods are nested inside of other functions, objects or classes. It may be that methods have to be nested inside of other code specifically.
 
-  Researched answer:
+  Researched answer: A method is an object property with a function value. A function is code that can be invoked to do a particular task.
```

Comment:
Nicely worded. Add a sentence to an answer like this of a recent time you used this tool. Then you can get into a conversation with your interviewer about your projects.